light/client.go

```diff
@@ -171,6 +186,11 @@ func NewClient(
 	options ...Option,
 ) (*Client, error) {
 
+	// Check that the witness list does not include duplicates or the primary
+	if err := validatePrimaryAndWitnesses(primary, witnesses); err != nil {
+		return nil, err
+	}
```

Comment:
We probably need to add this logic to `NewClientFromTrustedStore` as well

Reply:
Just ran statesync and it seems fine.